os/board/common/crashdump.c

```diff
@@ -57,8 +59,75 @@
  * Private Functions
  ****************************************************************************/
 
-#if defined(CONFIG_BOARD_RAMDUMP_UART)
-static int ramdump_via_uart(void)
+#if defined(CONFIG_BOARD_DUMP_UART)
+static int get_userfs_address_size(uint32_t *userfs_start, size_t *userfs_size)
+{
+	char flash_list[] = CONFIG_FLASH_PART_NAME;
+	char flash_sizes[] = CONFIG_FLASH_PART_SIZE;
+	char fs_part_name[] = "user";
+	char tmp_part_name[USERFS_PART_NAME_LEN + 1] = { '\0' };
+	size_t tmp_part_size = 0;
+	int i = 0;
+	int x = 0;
+	int c = 0;
+	*userfs_start = CONFIG_FLASH_START_ADDR;
+
+	while (flash_list[i] != '\0') {
+		if (flash_list[i] != 'u') {
+			while (flash_list[i] != ',') {
+				i++;
+			}
+			c++;
+		} else {
+			while ((x < (strlen(fs_part_name))) && (flash_list[i] != ',')) {
+				tmp_part_name[x] = flash_list[i];
+				x++;
+				i++;
+			}
+			tmp_part_name[x] = '\0';
+			if ((x == strlen(fs_part_name)) && (strncmp(tmp_part_name, fs_part_name, strlen(fs_part_name)) == 0)) {
```

Comment:
`strlen(fs_part_name)` is called three times.
1. Because it could be changed as definition, we could use the definition the size for that.
2. Or we could do once and could save in a variable and use it.

Reply:
Updated